homeassistant/components/keyboard_remote/__init__.py

```diff
@@ -165,6 +166,11 @@ def get_device_handler(self, descriptor):
             handler = self.handlers_by_descriptor[descriptor]
         elif dev.name in self.handlers_by_name:
             handler = self.handlers_by_name[dev.name]
+        else:
+            # check for symlinked paths matching descriptor
+            for d, h in self.handlers_by_descriptor.items():
+                if os.path.realpath(d) == descriptor:
```

Comment:
Could we normalize the stored paths instead so we don't need this extra check?

Reply:
So shouldn't we update the paths then?

Reply:
In fact the paths are already stored within the InputDevice objects, so an additional list is not needed.  I updated the check to make use of this and therefore reduce the number of os.path.realpath calls which are needed.